### PR TITLE
README & CircleCI config

### PR DESCRIPTION
**Pull request by tnolet**
As mentioned, you need to tweak some naming in the README.md and config.yml because there are pointers to github.com/tnolet/terraform-provider-checkly which should become ‘bitfield’. 

Maybe just search & replace ‘tnolet’.

Then just add this to a free Circle CI project and add a Github personal access token to the environment variables as GITHUB_TOKEN

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.10
+    working_directory: /go/src/github.com/tnolet/terraform-provider-checkly
```
Comment (bitfield) [inline .circleci/config.yml]: We may not actually need to set the working directory here; compare https://github.com/bitfield/uptimerobot/blob/master/.circleci/config.yml

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - checkout
+      - run: go get -v -t -d ./...
+      - run: go test -v ./...
+
+  deploy:
+    docker:
+      - image: circleci/golang:1.10
+    working_directory: /go/src/github.com/tnolet/terraform-provider-checkly
```
Comment (bitfield) [inline .circleci/config.yml]: Ditto

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.10
```
Comment (bitfield) [inline .circleci/config.yml]: Let's use `golang:1.12`, since that's the current release.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          name: cross compile
+          command: |
+            gox -os="linux darwin windows" -arch="amd64" -output="dist/terraform-provider-checkly_{{.OS}}_{{.Arch}}"
+            cd dist/ && gzip *
+      - add_ssh_keys
+      - run:
+          name: create release
+          command: |
+            if [ "$CIRCLE_TAG" ]; then
+              ghr -t $GITHUB_TOKEN -u tnolet -r terraform-provider-checkly --replace $CIRCLE_TAG dist/
```
Comment (bitfield) [inline .circleci/config.yml]: Presumably this needs to be `bitfield`.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# Checkly Terraform provider
+
+[![CircleCI](https://circleci.com/gh/tnolet/terraform-provider-checkly/tree/master.svg?style=svg)](https://circleci.com/gh/tnolet/terraform-provider-checkly/tree/master)
```
Comment (bitfield) [inline README.md]: If you update the path to 'gh/bitfield', I'll add it to my CircleCI config.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+This Terraform provider enables users to manage [Checkly](https://checklyhq.com) resources like checks.
+
+## Using the provider
+
+1. Download the binary applicable for your platform from the [latest tagged release](https://github.com/tnolet/terraform-provider-checkly/releases)
```
Comment (bitfield) [inline README.md]: Again let's make this 'bitfield'.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+```bash
+
+curl -OL https://github.com/tnolet/terraform-provider-checkly/releases/latest/download/terraform-provider-checkly_darwin_amd64.gz
+```
+
+2. Copy the binary to your Terraform plugin folder, unzip it and rename it to just `terraform-provider-checkly`
+
+```bash
```
Comment (bitfield) [inline README.md:15]: This is a bit fiddly; could we just include this script in the repo, and give people instructions to run a 'curl |sh'-type command?

File: README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+terraform providers
+.
+└── provider.checkly
+```
+
+If you're having issues, please check [the Hashicorp docs on installing third party plugins.](https://www.terraform.io/docs/configuration/providers.html#third-party-plugins)
+
+### Usage examples
+
+Add a `checkly_check` resource to your resource file
```
Comment (bitfield) [inline README.md]: We should document how to supply your API key via the provider config here.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+```terraform
+resource "checkly_check" "checkly-public-stats" {
```
Comment (bitfield) [inline README.md:58]: A more detailed example would be helpful here. In fact, what we could do is supply a 'minimal' example—just the basics to get a simple endpoint check working—and additionally a more featureful one which shows off everything Checkly does. For example, from the `test.tf` file:

```terraform
variable "checkly_api_key" {}

provider "checkly" {
  api_key = "${var.checkly_api_key}"
}

resource "checkly_check" "test-check2" {
  name             = "My test check 2"
  type             = "API"
  activated        = true
  should_fail      = true
  frequency        = 1
  ssl_check_domain = "example.com"
  double_check     = true

  locations = [
    "us-west-1",
    "ap-northeast-1",
    "ap-south-1",
  ]

  alert_settings {
    ssl_certificates {
      enabled         = true
      alert_threshold = 30
    }
  }

  request {
    follow_redirects = true
    url              = "http://example.com/"

    query_parameters = {
      search = "foo"
    }

    headers = {
      X-Bogus = "bogus"
    }

    assertion {
      source     = "JSON_BODY"
      property   = "code"
      comparison = "HAS_VALUE"
      target     = "authentication.failed"
    }

    assertion {
      source     = "STATUS_CODE"
      comparison = "EQUALS"
      target     = "401"
    }
  }
}
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+## Developing the provider
+
+Clone the repo, build the project and add it to the Terraform plugins directory. You will need to have Go installed.
+
+```bash
+git clone git@github.com:bitfield/terraform-provider-checkly.git
+cd terraform-provider-checkly
+go build
+go install
+cp terraform-provider-checkly ~/.terraform.d/plugins/darwin_amd64
```
Comment (bitfield) [inline README.md]: The `go install` and `copy` steps aren't necessary if you want to run Terraform inside the provider repo root. Terraform automatically looks for a plugin binary in the current directory. This is very useful when you're making changes, recompiling, and re-applying Terraform in a loop!

**Review by bitfield — CHANGES_REQUESTED** [review]
Great PR! Thanks a lot! If you update all the references to `tnolet` to `bitfield`, then I should be able to merge this more or less as-is.

File: README.md (hunk 9 above)
Comment (tnolet) [reply README.md:58]: I was looking for ways to generate documentation for TF providers, but didn't find anything. That is kind of unbelievable. There must be away to generate this from Go docs or something?

Comment (bitfield) [reply README.md:58]: You'd think!

Comment (tnolet) [reply README.md:58]: ha ha, well. Let me just resolve this one for now and I will dig up some nice examples and create a new PR just for that.

**Comment by tnolet** [comment]
All review comments are addressed in the last update except for the examples part. Let me chew on that for a couple of days.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+This Terraform provider enables users to manage [Checkly](https://checklyhq.com) resources like checks.
+
+## Using the provider
+
+1. Download the binary applicable for your platform from the [latest tagged release](https://github.com/bitfield/terraform-provider-checkly/releases). 
+Then copy the binary to your Terraform plugin folder, unzip it and rename it to just `terraform-provider-checkly`. Lastly, set the correct access rights.
+
+
+I you're on MacOS, just run the script `install.sh` script
```
Comment (bitfield) [inline README.md:13]: We could have this work for Linux and other platforms too, couldn't we, if the script selected the right binary for the kernel and architecture?

Comment (tnolet) [reply README.md:13]: Probably, for Linux anyway. For Windows I have no idea how that should work. Shall I add one just for Linux and let the Windows part as an exercise to the reader?

Comment (bitfield) [reply README.md:13]: Yes, I think MacOS and Linux are the low-hanging fruit; if we can make the install completely seamless for those users, that's a big win.

Comment (tnolet) [reply README.md:13]: Okido! Should be good in the last update.

**Review by tnolet — COMMENTED** [review]
should be good now

**Comment by tnolet** [comment]
@bitfield not sure what I'm missing, but the PR keeps mentioning one more requested change. Can you point me in the right direction?

**Comment by bitfield** [comment]
No idea, but it all looks good to me. Nice work!

**Pull request merged**